tests/test_verify_base_url.py

```diff
@@ -54,3 +60,21 @@ def test_disable_verify_via_env(testdir, httpserver, monkeypatch):
     httpserver.serve_content(content='<h1>Error!</h1>', code=500)
     result = testdir.runpytest('--base-url', httpserver.url)
     assert result.ret == 0
+
+
+def test_verify_retries_url(testdir, httpserver, monkeypatch):
+    testdir.makepyfile('def test_pass(): pass')
+    monkeypatch.setenv('VERIFY_BASE_URL', False)
+    monkeypatch.setattr(Retry, 'BACKOFF_MAX', 5)
+    httpserver.serve_content(content='<h1>Error!</h1>', code=500)
+    reprec = testdir.inline_run('--base-url', httpserver.url,
+                                '--verify-base-url')
+    passed, skipped, failed = reprec.listoutcomes()
+    assert len(failed) == 1
+    out = failed[0].longrepr.reprcrash.message
+    assert 'UsageError: Base URL failed verification!' in out
+    httpserver.serve_content(content='<h1>Loaded!</h1>')
+    reprec = testdir.inline_run('--base-url', httpserver.url,
+                                '--verify-base-url')
+    passed, skipped, failed = reprec.listoutcomes()
+    assert len(passed) == 1
```

Comment:
Could you add a test for verifying when the URL can't be resolved? Something like 'http://foo' would be fine. This one wouldn't need to recover.